doc/source/whatsnew/v1.5.0.rst

```diff
@@ -227,6 +227,7 @@ MultiIndex
 I/O
 ^^^
 - Bug in :meth:`DataFrame.to_stata` where no error is raised if the :class:`DataFrame` contains ``-np.inf`` (:issue:`45350`)
+- Bug in :meth:`read_excel` with `engine=python` results in an infinite loop with certain `skiprows` callables (:issue:`45585`)
```

Comment:
What does engine=python mean? I do not think that python is valid. Please also add ticks around the string, e.g. ``engine="python"``

Reply:
I'm not sure it was suggested here https://github.com/pandas-dev/pandas/pull/45586#discussion_r790897640

I've just removed it.

Reply:
Please let the reviewer resolve conversations

I think I get what jeff meant. ``read_csv with engine="python" and read_excel``. 

Please also add a test for read_csv